flake.nix

```diff
@@ -1,120 +1,50 @@
 {
-  description = "(insert short project description here)";
-
-  # Nixpkgs / NixOS version to use.
-  inputs.nixpkgs.url = "nixpkgs/nixos-20.09";
-
-  # Upstream source tree(s).
-  inputs.hello-src = { url = git+https://git.savannah.gnu.org/git/hello.git; flake = false; };
-  inputs.gnulib-src = { url = git+https://git.savannah.gnu.org/git/gnulib.git; flake = false; };
-
-  outputs = { self, nixpkgs, hello-src, gnulib-src }:
-    let
-
-      # Generate a user-friendly version numer.
-      version = builtins.substring 0 8 hello-src.lastModifiedDate;
-
-      # System types to support.
-      supportedSystems = [ "x86_64-linux" ];
-
-      # Helper function to generate an attrset '{ x86_64-linux = f "x86_64-linux"; ... }'.
-      forAllSystems = f: nixpkgs.lib.genAttrs supportedSystems (system: f system);
-
-      # Nixpkgs instantiated for supported system types.
-      nixpkgsFor = forAllSystems (system: import nixpkgs { inherit system; overlays = [ self.overlay ]; });
-
-    in
-
-    {
-
-      # A Nixpkgs overlay.
-      overlay = final: prev: {
-
-        hello = with final; stdenv.mkDerivation rec {
-          name = "hello-${version}";
-
-          src = hello-src;
-
-          buildInputs = [ autoconf automake gettext gnulib perl gperf texinfo help2man ];
-
-          preConfigure = ''
-            mkdir -p .git # force BUILD_FROM_GIT
-            ./bootstrap --gnulib-srcdir=${gnulib-src} --no-git --skip-po
+  description = "Deltabot : Bots manager for deltachat";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/master";
+    flake-utils.url = "github:numtide/flake-utils";
+    mach-nix.url = "github:DavHau/mach-nix?ref=3.3.0";
+    deltabot-scr = {
+      url = "https://files.pythonhosted.org/packages/c6/d7/90c24bbeef1e6b7e73dfe30e51e96699f5db05984497831ae2d87e93d9d4/deltabot-0.8.0.tar.gz";
+      flake = false;
+    };
+  };
+
+  outputs = { self, nixpkgs, flake-utils, mach-nix, deltabot-scr }:
+    flake-utils.lib.eachSystem [ "x86_64-linux" ] (system:
+      let
+        python = "python38";
+        pkgs = nixpkgs.legacyPackages.${system};
+        mach-nix-wrapper = import mach-nix { inherit pkgs python; };
+        requirements = ''
+          setuptools
+          attrs
+          deltachat
+          requests
+          pytest
+        '';
+        pythonBuild = mach-nix-wrapper.mkPython { inherit requirements; };
+        deltabot = mach-nix-wrapper.buildPythonPackage rec{
+          version = "0.8.0";
+          src = deltabot-scr;
+          requirementsExtra = requirements;
+          tests = true;
+          checkPhase = ''
+            export HOME=/tmp
```

Comment:
I don't think you are allowed to use the /tmp directory directly (in sandbox mode if I recall correctly).
You have to do something like `export HOME="$(mktemp -d)"`
also in the spirit of shell it's better to assign and export in a separate step (the reason is that if a subshell has a failure exporting will hide that failure).

Reply:
done as reviewed